### PR TITLE
feat: Migrate to Docusaurus v3

### LINKED ISSUE #9162: Website: Upgrade Docusaurus to the Latest Version (3.x)
The zio.dev website is based on docusaurus version 2.2.0. We need to update the site with the latest stable version.

### PR DESCRIPTION
**Pull request by neo773**
This PR migrates Docusaurus to v3

These PRs need to be merged first
https://github.com/zio/zio-logging/pull/881
https://github.com/zio/zio-quill/pull/3083
https://github.com/zio/zio-http/pull/3088
https://github.com/zio/zio-crypto/pull/293

https://github.com/user-attachments/assets/c13de178-120d-44b7-a341-882d28841425

/closes #9162
/claim #9162

### CODE REVIEW

**Comment by 987Nabil** [comment]
@neo773 CI is red for https://github.com/zio/zio-quill/pull/3083

**Comment by neo773** [comment]
@987Nabil 
Does not appears to be from the PR. They're all exiting with the same error code `/build/setup_databases.sh: line 6: docker-compose: command not found`

https://github.com/zio/zio-quill/actions/runs/10688428309/job/29655594684?pr=3083#step:5:77
https://github.com/zio/zio-quill/actions/runs/10688428309/job/29655593055?pr=3083#step:5:77
https://github.com/zio/zio-quill/actions/runs/10688428309/job/29655595488?pr=3083#step:5:190

**Comment by kyri-petrou** [comment]
@neo773 can you take a look at the current CI failure? Not entirely sure what's causing it

**Comment by neo773** [comment]
@kyri-petrou 
Some of the linked PRs haven't been merged yet and released on npm.
Build fails because the document breaks with MDX parser in Docusaurus v3

Check PR's body

**Comment by kyri-petrou** [comment]
> Some of the linked PRs haven't been merged yet and released on npm.

AFAICT all the PRs have been merged in. Or do we need to make releases for each library as well?

**Comment by neo773** [comment]
> > Some of the linked PRs haven't been merged yet and released on npm.
> 
> AFAICT all the PRs have been merged in. Or do we need to make releases for each library as well?

My bad, it's been a while since I made this PR.

Yes, you'll need to release a new version on npm since that's where the build process fetches the markdown files from.

**Comment by 987Nabil** [comment]
I think zio crypto might be dead. Maybe we should exclude it

**Comment by jdegoes** [comment]
@neo773 Can you remove zio-crypto? Would love to get Docusaurus upgraded!

cc @khajavi

**Comment by neo773** [comment]
`zio-quill` needs an npm release update, apart from that looks good to go

**Comment by khajavi** [comment]
> `zio-quill` needs an npm release update, apart from that looks good to go

@neo773  The npm release is not separate from the library release. So, can you file an issue on the zio-quill repo and ask the maintainers for the next release?

**Comment by kyri-petrou** [comment]
@neo773 can you also remove `zio-keeper`? Seems to be causing the CI failure and that project is not maintained (or ever had a stable release)

**Comment by neo773** [comment]
So I ran the build again this the last PR that needs to be merged for the migration to complete

https://github.com/zio/zio-dynamodb/pull/544

image

@kyri-petrou

**Comment by neo773** [comment]
@kyri-petrou 
CI is passing now, can we merge?

File: website/patch-guides.sh
Comment (khajavi) [inline website/patch-guides.sh:1]: It appears that something has gone wrong with the build file that we require this symlink.

Comment (neo773) [reply website/patch-guides.sh:1]: Yes, that specific guide in documentation folder broke and wasn’t detected during sidebar generation step

I did try looking at the build.sbt file but I’m not really familiar with Scala lang and it’s tooling.

if you have any suggestions for this I’d be happy to look into it.

Comment (khajavi) [reply website/patch-guides.sh:1]: Did this happen before or after you updated your work to align with the series/2.x branch?

Comment (neo773) [reply website/patch-guides.sh:1]: Happened after upgrading docusaurus to v3

**Comment by kyri-petrou** [comment]
@neo773 I'm okay to merge once @khajavi approves it :)

**Review by khajavi — APPROVED** [review]

**Pull request merged**
